benchmark/error/node-error.js

```diff
@@ -3,19 +3,21 @@
 const common = require('../common');
 
 const bench = common.createBenchmark(main, {
-  n: [1e7],
+  n: [1e6],
+  code: ['built-in', 'ERR_HTTP2_STREAM_SELF_DEPENDENCY'],
+  stackTraceLimit: [0, 10],
 }, {
   flags: ['--expose-internals'],
 });
 
-function main({ n }) {
-  const {
-    codes: {
-      ERR_INVALID_STATE,
-    },
-  } = require('internal/errors');
+
+function main({ n, code, stackTraceLimit }) {
+  const Err = code === 'built-in' ? Error : require('internal/errors').codes[code];
```

Comment:
To prevent dead code elimination making this benchmark invalid, the errors should be saved somewhere, for example fill an array with n errors, then replace that n errors with another n errors, preferably with the parameters passed into the error constructor varying in each iteration.

(If Turbofan is smart enough, `new ERR_HTTP2_STREAM_SELF_DEPENDENCY()` should probably be eliminated completely as dead code here).

Reply:
Also there's something else should be benchmarked - the performance of accessing `err.stack`. The stack is actually not computed until it's accessed.

Reply:
i applied @RafaelGSS suggestion with assert.ok

Reply:
I think an array might still be necessary - I've ran into cases where an operation that's too simple can still be discarded by Turbofan if you only assign it to a variable. The operation here incurs no side effect and it is always involving a constant. This is also all JS. It is theoretically still possible for Turbofan to skip the rest of the loop after OSR.

Reply:
Can you please provide me the code to apply.

Reply:
Can I just push to this branch?

Reply:
Sure, feel free. Thank you